Makefile

```diff
@@ -5,15 +5,56 @@
 # Please see GNU make's documentation if unfamiliar: https://www.gnu.org/software/make/manual/html_node/
 .PHONY: test build e2e dist clean format lint check site
 
+# Include versions of tools we build on-demand
+include Tools.mk
+
 # This should be driven by automation and result in N.N.N, not vN.N.N
 VERSION   ?= dev
 build_dir := build
 dist_dir  := dist
 
-# Build the path relating to the current runtime (goos,goarch)
-goos   := $(shell go env GOOS)
-goarch := $(shell go env GOARCH)
-goexe  := $(shell go env GOEXE)
+# This selects the goroot to use in the following priority order:
+# 1. ${GOROOT}          - Ex actions/setup-go
+# 2. ${GOROOT_1_17_X64} - Ex GitHub Actions runner
+# 3. $(go env GOROOT)   - Implicit from the go binary in the path
+#
+# There may be multiple GOROOT variables, so pick the one matching go.mod.
+go_release           := $(shell sed -ne 's/^go //gp' go.mod)
+# https://github.com/actions/runner/blob/master/src/Runner.Common/Constants.cs
+github_runner_arch   := $(if $(findstring $(shell uname -m),x86_64),X64,ARM64)
+github_goroot_name   := GOROOT_$(subst .,_,$(go_release))_$(github_runner_arch)
+github_goroot_val    := $(value $(github_goroot_name))
+# This works around missing variables on macOS via naming convention.
+# Ex. /Users/runner/hostedtoolcache/go/1.17.1/x64
+# Remove this after actions/virtual-environments#4156 is solved.
+github_goroot_cache  := $(lastword $(shell ls -d $(RUNNER_TOOL_CACHE)/go/$(go_release)*/$(github_runner_arch) 2>/dev/null))
+goroot_path          := $(shell go env GOROOT 2>/dev/null)
+goroot               := $(firstword $(GOROOT) $(github_goroot_val) $(github_goroot_cache) $(goroot_path))
+
+ifndef goroot
```

Comment:
added this because I was blind due to a typo in my Dockerfile until I added it :D